core/headerchain.go

```diff
@@ -423,10 +424,18 @@ func (hc *HeaderChain) setStateProcessing() bool {
 func (hc *HeaderChain) AppendBlock(block *types.WorkObject) error {
 	blockappend := time.Now()
 	// Append block else revert header append
-	logs, err := hc.bc.Append(block)
+	logs, unlocks, err := hc.bc.Append(block)
 	if err != nil {
 		return err
 	}
+	if unlocks != nil && len(unlocks) > 0 {
+		go func() {
```

Comment:
Why did you put this here?

Reply:
you asked to do it asynchronously

Reply:
This way guarantees the event will be sent once the channel has capacity, without blocking append

Reply:
```
for _, f := range filters[UnlocksSubscription] {
		select {
		case f.unlocks <- ev:
		default:
		}
	}
```
This is asynchronous, i.e. it does not wait for the consumer to respond before continuing. This is all we need to do, we don't need a goroutine here.

Reply:
but it does not guarantee the event gets sent. What I have is also non-blocking, but guarantees the event is sent.

Reply:
removed